core/cap-0057.md

```diff
@@ -916,7 +1039,32 @@ and a proof must be verified (see No Archival Fees for Entry Creation).
 is not currently stored by the validators. The operation can restore a mix of entries that
 are archived but currently stored by validators and those that are not stored by validators.
 Restoration proofs are included in the `proofs` struct inside `SorobanTransactionData` for
-the given transaction.
+the given transaction. Restore proof verification works as follows:
+
+```
+// readOnly must be empty
+for key in footprint.readWrite:
+    if key exists in live BucketList:
+        // Key is already live
+        continue
+
+    // Incudes hot archive, every pending archival snapshot, and cold archive
+    // Iteration is in order of most recent to oldest epoch
+    for bucketList in allBucketListsStoredOnDisk:
+        if ARCHIVED(key) exists in bucketList:
+            // Entry is archived, restore it
+            restore(key)
+            continueToNextKey
+        else if DELETED(key) exists in bucketList:
+            // Entry was most recently deleted, do not allow outdated restore
+            failTx()
+
+    // oldest epoch is epoch is cold archive (if one exists), the oldest pending archival snapshot,
```

Comment:
`oldest epoch is epoch is cold archive` - seems like incorrect wording and I'm not sure what the implied meaning here is